src/features/controls.ts

```diff
@@ -104,380 +104,375 @@ export interface ControlsInterface {
+    @property(
+        {type: String, attribute: 'camera-target', hasChanged: () => true})
+    cameraTarget: string = DEFAULT_CAMERA_TARGET;
 
-        @property(
-            {type: String, attribute: 'field-of-view', hasChanged: () => true})
-        fieldOfView: string = DEFAULT_FIELD_OF_VIEW;
+    @property(
+        {type: String, attribute: 'field-of-view', hasChanged: () => true})
+    fieldOfView: string = DEFAULT_FIELD_OF_VIEW;
 
-        @property({type: Number, attribute: 'interaction-prompt-threshold'})
-        interactionPromptThreshold: number =
-            DEFAULT_INTERACTION_PROMPT_THRESHOLD;
+    @property({type: Number, attribute: 'interaction-prompt-threshold'})
+    interactionPromptThreshold: number = DEFAULT_INTERACTION_PROMPT_THRESHOLD;
 
-        @property({type: String, attribute: 'interaction-prompt'})
-        interactionPrompt: InteractionPromptStrategy =
-            InteractionPromptStrategy.AUTO;
+    @property({type: String, attribute: 'interaction-prompt'})
+    interactionPrompt: InteractionPromptStrategy =
+        InteractionPromptStrategy.AUTO;
 
-        @property({type: String, attribute: 'interaction-policy'})
-        interactionPolicy: InteractionPolicy = InteractionPolicy.ALWAYS_ALLOW;
+    @property({type: String, attribute: 'interaction-policy'})
+    interactionPolicy: InteractionPolicy = InteractionPolicy.ALWAYS_ALLOW;
 
-        protected[$promptElement]: Element;
+    protected[$promptElement]: Element;
 
-        protected[$userPromptedOnce] = false;
-        protected[$waitingToPromptUser] = false;
-        protected[$shouldPromptUserToInteract] = true;
+    protected[$userPromptedOnce] = false;
+    protected[$waitingToPromptUser] = false;
+    protected[$shouldPromptUserToInteract] = true;
 
-        protected[$controls]: SmoothControls;
+    protected[$controls]: SmoothControls;
 
-        protected[$idealCameraDistance]: number|null = null;
-        protected[$lastSpherical] = new Spherical();
-        protected[$jumpCamera] = false;
+    protected[$idealCameraDistance]: number|null = null;
+    protected[$lastSpherical] = new Spherical();
+    protected[$jumpCamera] = false;
 
-        protected[$changeHandler] = (event: Event) =>
-            this[$onChange](event as ChangeEvent);
+    protected[$changeHandler] = (event: Event) =>
+        this[$onChange](event as ChangeEvent);
 
-        protected[$focusHandler] = () => this[$onFocus]();
-        protected[$blurHandler] = () => this[$onBlur]();
+    protected[$focusHandler] = () => this[$onFocus]();
+    protected[$blurHandler] = () => this[$onBlur]();
 
-        protected[$promptTransitionendHandler] = () =>
-            this[$onPromptTransitionend]();
+    protected[$promptTransitionendHandler] = () =>
+        this[$onPromptTransitionend]();
 
-        constructor() {
-          super();
-          const scene = (this as any)[$scene];
+    constructor(...args: any[]) {
+      super(...args);
```

Comment:
What are these new args?

Reply:
The stricter checks of the Google TypeScript compiler mark this as a case where we are using a mixin pattern. The rule for mixin class extension like this is that constructor args _must_ be blanket forwarded to the super constructor.

Reply:
I see, maybe my question would be better expressed as why does only this Mixin seem to have a constructor? Is it even necessary?

Reply:
No, it probably isn't. I think some of this stuff could be re-written as class field initializers, and the rest ought to be moved to `connectedCallback`.

Reply:
I factored out the explicit constructor, so now this chunk of implementation is a bit cleaner.